extensions/sql/contract-definition/loader/src/test/java/org/eclipse/dataspaceconnector/sql/contractdefinition/loader/SqlContractDefinitionLoaderTest.java

```diff
@@ -0,0 +1,101 @@
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.eclipse.dataspaceconnector.sql.SqlQueryExecutor.executeQuery;
+
+class SqlContractDefinitionLoaderTest extends AbstractSqlContractDefinitionLoaderTest {
+
+    private static final String DATASOURCE_NAME = "contractdefinition";
+
+    private final Map<String, String> systemProperties = new HashMap<>() {
+        {
+            put(String.format("edc.datasource.%s.url", DATASOURCE_NAME), "jdbc:h2:mem:test");
+            put(String.format("edc.datasource.%s.driverClassName", DATASOURCE_NAME), org.h2.Driver.class.getName());
+            put("edc.contractdefinition.datasource.name", DATASOURCE_NAME);
+        }
+    };
+
+    @BeforeEach
+    void setUp() {
+        systemProperties.forEach(System::setProperty);
+    }
+
+    @AfterEach
+    void tearDown() {
+        getTransactionContext().execute(() -> {
+            try (Connection connection = getDataSourceRegistry().resolve(DATASOURCE_NAME).getConnection()) {
+                executeQuery(connection, String.format("DELETE FROM %s", SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE));
+            } catch (SQLException e) {
+                throw new RuntimeException(e);
+            }
+        });
+        systemProperties.keySet().forEach(System::clearProperty);
+    }
+
+    @Test
+    @DisplayName("Context Loads, tables exist")
+    void contextLoads() throws SQLException {
+        String query = String.format("SELECT 1 FROM %s", SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE);
+        executeQuery(getDataSourceRegistry().resolve(DATASOURCE_NAME).getConnection(), query);
+    }
+
+    @Test
+    @DisplayName("Accepts and persists ContractDefinition")
+    void testAcceptEntry() throws SQLException {
+        Policy accessPolicy = Policy.Builder.newInstance()
+                .id("access")
+                .build();
+        Policy contractPolicy = Policy.Builder.newInstance()
+                .id("contract")
+                .build();
+        AssetSelectorExpression expression = AssetSelectorExpression.Builder.newInstance()
+                .build();
+        ContractDefinition definition = ContractDefinition.Builder.newInstance()
+                .id("id")
+                .accessPolicy(accessPolicy)
+                .contractPolicy(contractPolicy)
+                .selectorExpression(expression)
+                .build();
+
+        getContractDefinitionLoader().accept(definition);
+
+        String query = String.format("SELECT count(*) FROM %s WHERE %s=?",
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_ID);
+
+        Long definitionCount = executeQuery(getDataSourceRegistry().resolve(DATASOURCE_NAME).getConnection(),
+                (rs) -> rs.getLong(1),
+                query, "id").iterator().next();
+
+        Assertions.assertEquals(1, definitionCount);
```

Comment:
Just as a side note, we're using AssertJ, that would make this `assertThat(definitionCount).isEqualTo(1)`